services/src/bin/main.rs

```diff
@@ -1,55 +1,93 @@
-use flexi_logger::writers::{FileLogWriter, FileLogWriterHandle};
+use flexi_logger::writers::FileLogWriter;
 use flexi_logger::{Age, Cleanup, Criterion, FileSpec, Naming, WriteMode};
 use geoengine_services::error::Result;
 use geoengine_services::util::config;
 use geoengine_services::util::config::get_config_element;
 use std::str::FromStr;
 use time::format_description;
 use tracing::metadata::LevelFilter;
+use tracing::Subscriber;
 use tracing_subscriber::prelude::*;
+use tracing_subscriber::registry::LookupSpan;
+use tracing_subscriber::Layer;
 
 #[tokio::main]
 async fn main() {
-    let _log_writer_handle = initialize_logging();
-
     start_server().await.unwrap();
 }
 
 #[cfg(not(feature = "pro"))]
 pub async fn start_server() -> Result<()> {
+    reroute_gdal_logging();
+    let logging_config: config::Logging = get_config_element()?;
+    let registry = tracing_subscriber::registry().with(console_layer(&logging_config));
+    let registry = registry.with(file_layer(&logging_config));
+    registry.init();
+
     geoengine_services::server::start_server(None).await
 }
 
 #[cfg(feature = "pro")]
 pub async fn start_server() -> Result<()> {
-    geoengine_services::pro::server::start_pro_server(None).await
-}
+    use opentelemetry_jaeger::config::agent::AgentPipeline;
+    use tracing_opentelemetry::OpenTelemetryLayer;
 
-/// Tries to initialize `tracing-subscriber` logging, `flexi-logging` and `GDAL` logging.
-///
-/// # Panics
-/// If anything cannot be initialized, this method panics as it is supposed to run at startup.
-///
-fn initialize_logging() -> Option<FileLogWriterHandle> {
     reroute_gdal_logging();
+    let logging_config: config::Logging = get_config_element()?;
+    let open_telemetry_config: geoengine_services::pro::util::config::OpenTelemetry =
+        get_config_element()?;
+
+    let registry = tracing_subscriber::registry();
+
+    let open_telemetry_layer = if open_telemetry_config.enabled {
+        let tracer = AgentPipeline::default()
+            .with_endpoint(open_telemetry_config.endpoint)
+            .with_service_name("Geo Engine")
+            .install_simple()?;
+        let opentelemetry: OpenTelemetryLayer<tracing_subscriber::Registry, _> =
+            tracing_opentelemetry::layer().with_tracer(tracer);
+        Some(opentelemetry)
+    } else {
+        None
+    };
+
+    let registry = registry.with(open_telemetry_layer);
+    let registry = registry.with(console_layer(&logging_config));
+    let registry = registry.with(file_layer(&logging_config));
+    registry.init();
```

Comment:
1. what do you mean? this is already a function with the whole purpose of setting up the logging and then calling the actual start method. If I extract it into another method it would be exactly the same.
2. this is not a service but a global logging/tracing subscriber. How does this to relate to actix services and routing?